UIPS/UIP-0131.md

```diff
@@ -0,0 +1,121 @@
+
+5.  Point becomes active on junior chain once minimum confirmation period is met
+
+**Return Process**
+
+1.  Point owner initiates return process through the Embassy contract
+
+2.  Embassy contract creates a return ticket containing destination address and merkle root of all junior chain activity
+
+3.  Galaxy oracles validate the return ticket
+
+4.  Ministry contract verifies oracle signatures
+
+5.  Ministry contract releases point from the vault with specified state
+
+### State Management
+
+The Embassy contract maintains just the current state of points on its chain---their ownership addresses and proxy configurations. When a point arrives through the Ministry bridge, the Embassy registers only this essential current state data.
+
+For returns, the oracle-signed tickets need only contain the point's current ownership address and proxy configurations. This lightweight approach leverages the oracle trust model---since oracles stake their galaxies on providing accurate state transitions, maintaining historical state becomes unnecessary. The optimistic rollup's challenge mechanism relies on oracle signatures rather than state history verification.
+
+### Oracle Security Model
+
+Galaxies secure Ministry by staking their own identities in the contract. To operate as an oracle, a galaxy transfers itself to the Ministry contract as collateral. This ensures stake value inherently exceeds typical transfer values, as galaxy ownership is generally more valuable than subordinate points. While certain points may accumulate exceptional value through smart contract integrations or attached assets that could theoretically exceed a galaxy's worth, this economic reality does not impair the protocol's security model.
+
+The system enforces correct oracle behavior through social consensus and economic incentives. Invalid transfer tickets serve as evidence of misbehavior---any party can submit transaction references showing fraudulent ticket validation to the Ministry contract. These references locate the original transfer records on each chain. The reports support Senate governance decisions which may result in galaxy forfeiture. Multiple oracles provide redundancy in signature collection and transfer validation, ensuring the bridge remains operational even if some oracles are temporarily offline.
+
+The protocol implements rate limiting, mandatory cooldowns, and minimum confirmation periods to prevent timing attacks. A monitoring client watches both chains for fraudulent tickets and automates violation reporting.
+
+TODO: Specify signature thresholds, timeouts, rate limits, and violation report format.
+
+### Arvo Integration
+
+The introduction of multi-chain state requires significant changes to Urbit's PKI verification system. Currently, %jael maintains Azimuth state by synthesizing data from Ethereum and the naive rollup. The addition of Base requires extending this system to handle three distinct sources of truth.
```

Comment:
> %jael maintains Azimuth state by synthesizing data from Ethereum and the naive rollup.

Correction: this synthesizing happens in `/app/azimuth`. Jael itself simply listens to that (and is trivially configured to listen to a different userspace agent) on a much smaller interface surface: jael only hears about changes to networking-relevant PKI state (namely public keys w/ revision nrs, breach nrs, sponsors).

(`/app/azimuth` in turn uses `/app/eth-watcher`, which is almost fully generic, for both azimuth.eth and naive rollup inspection. The codepaths for those two are so similar that it wouldn't be crazy to say that from arvo's perspective, these are still one source of truth. Certainly they differ much less from each other than any new addition like Base would differ from them.)